src/test/kotlin/net/atos/zac/app/klant/model/personen/RestPersoonTest.kt

```diff
@@ -0,0 +1,136 @@
+package net.atos.zac.app.klant.model.personen
+
+import io.kotest.core.spec.style.BehaviorSpec
+import io.kotest.matchers.shouldBe
+import net.atos.client.brp.model.generated.AbstractDatum
+import net.atos.client.brp.model.generated.OpschortingBijhouding
+import net.atos.client.brp.model.generated.PersoonInOnderzoek
+import net.atos.client.brp.model.generated.PersoonInOnderzoekBeperkt
+import net.atos.client.brp.model.generated.RniDeelnemer
+import net.atos.client.brp.model.generated.Waardetabel
+import java.util.EnumSet
+
+class RestPersoonTest : BehaviorSpec({
+
```

Comment:
The tests below could possibly simplified by using a `withData`, to do [data diven testing](https://kotest.io/docs/framework/datatesting/data-driven-testing.html), what do you think?

Reply:
Not really. All the construction and checks has to be the repeated in the same way as `Persoon` and `PersoonBeperkt` do not share a common parent.

Instead of repeating Given-When-Then we'll have repeating construction of objects outside of Given block, which do not make things more readable.

Unfortunately in the current format this one is not appropriate for data driven testing. If only we had a proper inheritance (or composition even) this would have been different.